src/v2/Apps/Order/Components/__tests__/AddressModal.jest.tsx

```diff
@@ -24,40 +25,112 @@ const savedAddress: SavedAddressType = {
   isDefault: false,
 }
 
-const testAddressModalProps: Props = {
-  show: true,
-  address: savedAddress,
-  onSuccess: jest.fn(),
-  onError: jest.fn(),
-  onDeleteAddress: jest.fn(),
-  modalDetails: {
-    addressModalTitle: "Modal title",
-    addressModalAction: "editUserAddress",
-  },
-  me: {
-    id: "1234",
-    addressConnection: {
-      edges: [],
-    },
-    " $refType": "SavedAddresses_me",
-  },
-  closeModal: jest.fn(),
-}
+let testAddressModalProps: Props
 
 function getWrapper(props: Props) {
   return mount(<AddressModal {...props} />)
 }
 
 describe("AddressModal", () => {
   beforeEach(() => {
+    testAddressModalProps = {
+      show: true,
+      address: savedAddress,
+      onSuccess: jest.fn(),
+      onError: jest.fn(),
+      onDeleteAddress: jest.fn(),
+      modalDetails: {
+        addressModalTitle: "Edit address",
+        addressModalAction: "editUserAddress",
+      },
+      me: {
+        id: "1234",
+        addressConnection: {
+          edges: [],
+        },
+        " $refType": "SavedAddresses_me",
+      },
+      closeModal: jest.fn(),
+    }
     commitMutation.mockReset()
   })
-  it("renders Modal with the title and input fields", () => {
+  it("renders EditModal with the title, input fields and buttons", () => {
     const wrapper = getWrapper(testAddressModalProps)
-    expect(wrapper.text()).toContain("Modal title")
+    expect(wrapper.text()).toContain("Edit address")
+    expect(wrapper.find("input").length).toBe(7)
+    expect(wrapper.find("select").length).toBe(1)
+    expect(wrapper.find("Text[data-test='deleteButton']").length).toBe(1)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
+  })
+
+  it("renders AddModal with the title, input fields and button", () => {
+    const wrapper = getWrapper({
+      ...testAddressModalProps,
+      modalDetails: {
+        addressModalTitle: "Add address",
+        addressModalAction: "createUserAddress",
+      },
+    })
+    expect(wrapper.text()).toContain("Add address")
     expect(wrapper.find("input").length).toBe(7)
     expect(wrapper.find("select").length).toBe(1)
+    expect(wrapper.find("Text[data-test='deleteButton']").length).toBe(0)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
+  })
+
+  it("clicking the delete button spawns a correct dialog", () => {
+    const wrapper = getWrapper(testAddressModalProps)
+    const deleteButton = wrapper.find("Text[data-test='deleteButton']")
+    const dialog = wrapper.find(Dialog)
+    expect(dialog.props().show).toBe(false)
+    deleteButton.simulate("click")
+
+    setTimeout(() => {
+      expect(dialog).toHaveLength(1)
+      expect(dialog.props().title).toBe("Delete address?")
+      expect(dialog.props().detail).toBe(
+        "This will remove this address from your saved addresses."
+      )
+      expect(dialog.props().primaryCta.text).toContain("Delete")
+      expect(dialog.props().secondaryCta.text).toContain("Cancel")
```

Comment:
non-blocking

I like the pattern we used above for looking for text on a component -- `expect(wrapper.text()).toContain("Edit address")` -- it is a bit more robust in that there are less chances for the test to break due to an implementation change like renaming a prop. 

Again, I'm not suggesting this change must be made to merge the PR; just suggesting that avoiding checking of `props` directly in an assertion eliminates one more surface for tests breaking due to an unimportant change.